.github/workflows/release.yml

```diff
@@ -58,5 +60,11 @@ jobs:
         run: |
           jfrog config add repox --artifactory-url $REPOX_URL --access-token $PROMOTE_ACCESS_TOKEN
           jfrog config use repox
-          jfrog rt bpr --status it-passed $PACKAGE $RELEASE_TAG sonarsource-npm-public-builds 
-          jfrog rt bpr --status released $PACKAGE $RELEASE_TAG sonarsource-npm-public-releases
+          jfrog rt bpr --status released ${{ github.event.repository.name }} ${{ steps.get_version.outputs.build }} sonarsource-npm-public-releases
+      - name: Publish npm package to npmjs
+        working-directory: ${{ steps.local_repo.outputs.dir }}
+        env:
+          NPM_TOKEN: ${{ fromJSON(steps.secrets.outputs.vault).npm_token }}
+        run: |
+          file=$(find -name "*.tgz" -printf "%f")
+          npm publish "$file"
```

Comment:
by default, npm publish will use the version that comes from the package.json.

because our jfrog artifact injects the build #, it is by default included when publishing
(see downloaded package.json)

```
{
  "name": "sonarqube-scanner",
  "description": "SonarQube/SonarCloud Scanner for the JavaScript world",
  "version": "4.2.0-2297",
   ...
```

https://repox.jfrog.io/ui/repos/tree/General/sonarsource-npm-public-dev/sonarqube-scanner/-/sonarqube-scanner-4.2.0-2297.tgz?clearFilter=true


If this is an issue, we can look at overriding the version to exclude the build # (but I think at Sonar by default we do publish the build version as well. see sonar-scanner-cli https://github.com/SonarSource/sonar-scanner-cli/releases/tag/6.1.0.4477)

options:
https://docs.npmjs.com/updating-your-published-package-version-number

Reply:
Well spotted.. Definitively looks like we don't want to publish versions with the build number on Repox IMO

Reply:
see https://github.com/SonarSource/sonar-scanner-npm/pull/159#discussion_r1701711790